arangodb/src/main/java/com/yahoo/ycsb/db/ArangoDBClient.java

```diff
@@ -0,0 +1,488 @@
+/*
+ * Arango client binding for YCSB.
+ *
+ * By Yemin, Sui
+ */
+package com.yahoo.ycsb.db;
+
+import com.arangodb.ArangoConfigure;
+import com.arangodb.ArangoDriver;
+import com.arangodb.ArangoException;
+import com.arangodb.ArangoHost;
+import com.arangodb.DocumentCursor;
+import com.arangodb.ErrorNums;
+import com.arangodb.entity.BaseDocument;
+import com.arangodb.entity.CollectionOptions;
+import com.arangodb.entity.DocumentEntity;
+import com.arangodb.entity.EntityFactory;
+import com.arangodb.entity.TransactionEntity;
+import com.arangodb.util.MapBuilder;
+
+import com.yahoo.ycsb.DB;
+import com.yahoo.ycsb.Status;
+import com.yahoo.ycsb.DBException;
+import com.yahoo.ycsb.ByteIterator;
+import com.yahoo.ycsb.StringByteIterator;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.Vector;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This's thread-safe.
+ * Assume the table is the same during whole procedure.
+ * The data will be transfered from ByteIterator into String before sent to DB.
+ * 
+ * @author ysui
+ */
+public class ArangoDBClient extends DB {
+
+  private static Logger logger = LoggerFactory.getLogger(ArangoDBClient.class);
+  
+  /**
+   * The database name to access.
+   */
+  private static String databaseName = "ycsb";
+
+  /**
+   * Count the number of times initialized to teardown on the last
+   * {@link #cleanup()}.
+   */
+  private static final AtomicInteger INIT_COUNT = new AtomicInteger(0);
+
+  /** ArangoDB Driver related, Singleton. */
+  private static ArangoConfigure arangoConfigure;
+  private static ArangoDriver arangoDriver;
+  private static Boolean dropDBBeforeRun;
+  private static Boolean waitForSync = false;
+  private static Boolean transactionUpdate = false;
+  // For creating collection manually with waitForSync param.
+  private static String defaultCollection = null;
+
+  /**
+   * Initialize any state for this DB. Called once per DB instance; there is
+   * one DB instance per client thread.
+   * 
+   * Actually, one client process will share one DB instance here.(Coincide to
+   * mongoDB driver)
+   */
+  @Override
+  public void init() throws DBException {
+    INIT_COUNT.incrementAndGet();
+    synchronized (ArangoDBClient.class) {
+      if (arangoDriver != null || arangoConfigure != null) {
```

Comment:
should this be an and? what if creating the driver object throws a runtime exception?